### PR TITLE
collect system metrics to mlflow, fix the bug related to collecting runs for the inference process

### PR DESCRIPTION
**Pull request by Anna050689**
collect system metrics to mlflow, fix the bug related to collecting runs for the inference process

### CODE REVIEW

File: src/syngen/ml/mlflow_tracker/mlflow_tracker.py [hunk 1]
```diff
@@ -90,18 +101,53 @@ def __new__(cls, experiment_name=None, is_active=False):
...
+            except (TypeError, StatisticsError):
+                pass
+
+    def collect_metrics(self, table, stage):
+        """
+        Collect the duration of the run
+        and hardware metrics related to the run
+        """
+        runs = self.search_runs(table_name=table, type_of_process=stage)
+        if runs:
```
[the remark this answers is not in the document]
Comment (Anna050689) [reply src/syngen/ml/mlflow_tracker/mlflow_tracker.py:142]: if runs == [], this condition will also work

File: src/syngen/ml/strategies/strategies.py [hunk 2]
```diff
@@ -197,13 +200,15 @@ def add_handler(self, type_of_process: str):
...
     def add_reporters(self):
-        if self.config.print_report or self.config.get_infer_metrics:
-            accuracy_reporter = AccuracyReporter(
-                table_name=self.config.table_name,
-                paths=self.config.paths,
-                config=self.config.to_dict(),
-            )
-            Report().register_reporter(table=self.config.table_name, reporter=accuracy_reporter)
+        table_name = self.config.table_name
+        if not table_name.endswith("_fk") and \
```
[the remark this answers is not in the document]
Comment (Anna050689) [reply src/syngen/ml/strategies/strategies.py:204]: This change has been provided to generate the report just for final table, not for surrogate which have endings - "_pk", "_fk"

File: src/syngen/ml/strategies/strategies.py [hunk 3]
```diff
@@ -226,12 +231,11 @@ def run(self, **kwargs):
                 log_level=kwargs["log_level"],
                 both_keys=kwargs["both_keys"],
             )
-            type_of_process = kwargs["type_of_process"]
-            if type_of_process == "infer":
```
[the remark this answers is not in the document]
Comment (Anna050689) [reply src/syngen/ml/strategies/strategies.py:230]: To log parameters in the run of the inference process during training

File: src/syngen/ml/mlflow_tracker/mlflow_tracker.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rioman) [reply src/syngen/ml/mlflow_tracker/mlflow_tracker.py:142]: Yes, but it's unnecessary

File: src/syngen/ml/strategies/strategies.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (rioman) [reply src/syngen/ml/strategies/strategies.py:230]: Was this an error earlier?

**Review by rioman — APPROVED** [review]

**Pull request merged**